### PR TITLE
LibGUI: Fix annoying slider imperfections

### PR DESCRIPTION
**Pull request by andreao**

### CODE REVIEW

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibGUI/Slider.cpp [hunk 1]
```diff
@@ -82,7 +82,9 @@ Gfx::IntRect Slider::knob_rect() const
 
 void Slider::start_drag(Gfx::IntPoint start_position)
 {
-    VERIFY(!m_dragging);
+    if (m_dragging) { // In case we lost the mouseup event
+        return;
```
[the remark this answers is not in the document]
Comment (andreao) [reply Userland/Libraries/LibGUI/Slider.cpp:86]: Right, I'll write it behind my ear and leave out the commit.

[the remark this answers is not in the document]
Comment (andreao) [reply Userland/Libraries/LibGUI/Slider.cpp]: In some other UI frameworks there is the concept of mouse enter/leave. Would that be an idea for a better solution in LibGUI?

[the remark this answers is not in the document]
Comment (awesomekling) [reply Userland/Libraries/LibGUI/Slider.cpp]: There are already mouse enter/leave events. (See `Widget::enter_event` and `Widget::leave_event`)

Re: writing it behind your ear, please open an issue about it instead. Then we don't have to remember to look behind your ear :)

[the remark this answers is not in the document]
Comment (andreao) [reply Userland/Libraries/LibGUI/Slider.cpp]: Don't know how I missed those, thanks :)
Agree about opening an issue instead as my ears have a tendency to get washed every morning... but even better to try to fix it myself! Your intuition was right that we should track this down, as it turns out to be caused by fake mouse events sent from `WindowManager.reevaluate_hover_state_for_window`, which causes dragging to be interrupted. Just gotta figure out how to best fix it :)

[the remark this answers is not in the document]
Comment (andreao) [reply Userland/Libraries/LibGUI/Slider.cpp]: Opened issue to get some feedback: https://github.com/SerenityOS/serenity/issues/17695

**Pull request merged**